dev-docs/source/RunningSanitizers.rst

```diff
@@ -0,0 +1,156 @@
+.. _RunningSanitizers:
+
+##################
+Running Sanitizers
+##################
+
+.. contents::
+    :local:
+
+Overview
+=========
+
+Sanitizers allow developers to find any issues with their code at runtime
+related to memory, threading or undefined behaviour.
+
+Pre-requisites
+==============
+
+The following tooling is required:
+
+- CMake 13.0 onwards (instructions for Ubuntu `here <https://apt.kitware.com/>`__)
+- GCC-8 onwards
```

Comment:
We should probably add something describing how to get gcc-8 on things like Ubuntu. Do we know what versions of clang this will work with?

Reply:
Should we add a check in cmake and bail out if gcc < 8 and the sanitizers are enabled? This means developers won't get too far with an incorrect compiler.

Reply:
Clang version documentation is a bit spread out, but it looks like 3.2 onwards is what we need

Looking at the c++ language support all clang version which can support any c++-17 features are compatible so I won't add an extra check for clang

I've added one for GCC